### PR TITLE
Remove CircleCI service already replaced by GitHub Actions CI

### PR DESCRIPTION
**Pull request by sandcha**
⚠️ When this PR is merged, we must also:
- [x]  update [this repository `master` branch settings 🔐](https://github.com/openfisca/openfisca-survey-manager/settings/branch_protection_rules/3856668) to remove `ci/circleci` checks from the required CI checks, 
- [x] set required checks to GitHub Actions jobs 
- [x] and clean up CircleCI website configuration (remove this repository from followed repositories ? ...): in [CircleCI](https://app.circleci.com/projects/project-dashboard/github/openfisca/), §Projects, Unfollowed `openfisca-survey-manager`.

- - - -

#### Technical changes

- Remove old `CircleCI` continuous integration configuration
  - Get `master` branch identification in `is-version-number-acceptable.sh` from GitHub Actions environment
- Set `README` CI badge to current `GitHub Actions` CI

- - - -

#### Additional information

The first commit comes from current `WIP : CI` pull request https://github.com/openfisca/openfisca-survey-manager/pull/238.

### CODE REVIEW

File: .python-version
Comment (benoit-cty) [inline .python-version:1]: Est-ce que l'on souhaite commit ce fichier dans le projet ?
Il est lié à quel outil, pyenv ?

File: openfisca_survey_manager/__init__.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 with open(os.path.join(test_config_files_directory, 'config.ini'), "w+") as file:
     file.write(config_ini)
 
-# GitHub actions test
+# GitHub Actions test
 if 'CI' in os.environ:
```
Comment (benoit-cty) [inline openfisca_survey_manager/__init__.py:47]: En une ligne :
```suggestion
 is_in_ci = 'CI' in os.environ
```

File: openfisca_survey_manager/tests/test_add_survey_to_collection.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from openfisca_survey_manager.scripts.build_collection import add_survey_to_collection
 from openfisca_survey_manager.input_dataframe_generator import set_table_in_survey
 
-# Travis, Gitlab runner, Gihub Action and circle has env variable "CI" set by default
+# GitLab Runner and GitHub Actions have env variable "CI" set by default
 if 'CI' in os.environ:
```
Comment (benoit-cty) [inline openfisca_survey_manager/tests/test_add_survey_to_collection.py:13]: En une ligne :
```suggestion
is_in_ci = 'CI' in os.environ
```

File: .python-version
Comment (sandcha) [reply .python-version:1]: Oops, this wasn't pushed on purpose 😅 
Yes, it's a pyenv configuration file that sets the Python version for the current repository. The file is already gitignored in #261 so this issue shouldn't occur twice.

File: openfisca_survey_manager/__init__.py (hunk 1 above)
Comment (sandcha) [reply openfisca_survey_manager/__init__.py]: Done in 430b7875394cadbf842526e2798be43b6f839e5a (to update the `else` line under the suggestion).

File: openfisca_survey_manager/tests/test_add_survey_to_collection.py (hunk 2 above)
Comment (sandcha) [reply openfisca_survey_manager/tests/test_add_survey_to_collection.py]: Done in 430b7875394cadbf842526e2798be43b6f839e5a (to update the `else` line under the suggestion).

File: .python-version
Comment (sandcha) [reply .python-version:1]: File removed from commit.

**Comment by sandcha** [comment]
Fixing lint and rebasing.

**Review by benoit-cty — APPROVED** [review]
Thanks !

**Comment by sandcha** [comment]
Removed CircleCI webhook : 

Capture d’écran 2023-07-20 à 16 42 37
Capture d’écran 2023-07-20 à 16 42 54
Capture d’écran 2023-07-20 à 16 43 07

**Pull request merged**